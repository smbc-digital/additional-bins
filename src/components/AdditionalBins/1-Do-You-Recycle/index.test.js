import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import getPageRoute from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'
import { DoYouRecycle } from './index'

Enzyme.configure({ adapter: new Adapter() })

describe('DoYouRecycle', () => {
	it('should push to correct page when answer is No, Not enought space', () => {
        // Arrange
		const history = { push: jest.fn() }
        const context = {
            doYouRecycle: {
                value: 'not-enough-space',
                isValid : true
            }
        }

		// Act
        const wrapper = mount(<DoYouRecycle history={history} context={context} />)
    
        // Assert
        wrapper.find('form').simulate('submit')
        expect(history.push).toHaveBeenCalledWith(getPageRoute(8))
    })
    
    it('should push to correct page when answer is No, What can i recycle', () => {
        // Arrange
        window.location.assign = jest.fn()
		const history = { push: jest.fn() }
        const context = {
            doYouRecycle: {
                value: 'what-can-i-recycle',
                isValid : true
            }
        }

		// Act
        const wrapper = mount(<DoYouRecycle history={history} context={context} />)
    
        // Assert
        wrapper.find('form').simulate('submit')
        expect(history.push).toHaveBeenCalledWith(getPageRoute(7))
    })
    
    it('should push to correct page when answer is Yes', () => {
		// Arrange
		const history = { push: jest.fn() }
        const context = {
            doYouRecycle: {
                value: 'true',
                isValid : true
            }
        }

		// Act
        const wrapper = mount(<DoYouRecycle history={history} context={context} />)
    
        // Assert
        wrapper.find('form').simulate('submit')
        expect(history.push).toHaveBeenCalledWith(getPageRoute(2))
    })
    
    it('should check if button disabled by default',() => { 
		// Arrange
		const context = {
            doYouRecycle: {
                value: '',
                isValid : false
            }
		}
		// Act
		const wrapper = mount(<DoYouRecycle context={context} history={history}/>)
		
		// Assert
		expect(wrapper.find('button').props().disabled).toBeTruthy()
})

    describe('snapshot', () => {
        it('renders correctly', () => {
            const context = {
                doYouRecycle: {
                    value: '',
                    isValid : false
                }
            }

            const tree = renderer
                .create(<DoYouRecycle context={context} />)
                .toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
