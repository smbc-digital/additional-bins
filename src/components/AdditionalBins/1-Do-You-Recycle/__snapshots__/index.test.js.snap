// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoYouRecycle snapshot renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Request an additional black bin
  </h1>
  <h2>
    Do you recycle as much as possible?
  </h2>
  <fieldset
    aria-label={undefined}
    role="radiogroup"
  >
    <legend
      className="sr-only"
    >
      Do you recycle as much as possible?
    </legend>
    <div
      className="input-radio-container"
    >
      <input
        checked={false}
        disabled={undefined}
        id="not-enough-space"
        name="doYouRecycle"
        onChange={[Function]}
        type="radio"
        value="not-enough-space"
      />
      <label
        htmlFor="not-enough-space"
      >
        No, I want to but there's not enough space in my recycling bins
      </label>
    </div>
    <div
      className="input-radio-container"
    >
      <input
        checked={false}
        disabled={undefined}
        id="what-can-i-recycle"
        name="doYouRecycle"
        onChange={[Function]}
        type="radio"
        value="what-can-i-recycle"
      />
      <label
        htmlFor="what-can-i-recycle"
      >
        No, I don't know what I can recycle
      </label>
    </div>
    <div
      className="input-radio-container"
    >
      <input
        checked={false}
        disabled={undefined}
        id="Yes"
        name="doYouRecycle"
        onChange={[Function]}
        type="radio"
        value="true"
      />
      <label
        htmlFor="Yes"
      >
        Yes
      </label>
    </div>
  </fieldset>
  <button
    className="button-primary"
    disabled={true}
    id={undefined}
    onClick={undefined}
    type="submit"
  >
    Next step
  </button>
</form>
`;
